src/main/java/net/geforcemods/securitycraft/tileentity/RetinalScannerTileEntity.java

```diff
@@ -67,4 +83,70 @@ public boolean activatedOnlyByPlayer() {
 		return new Option[]{ activatedByEntities, sendMessage };
 	}
 
+	public static void setProfileCache(PlayerProfileCache profileCacheIn) {
+		profileCache = profileCacheIn;
+	}
+
+	public static void setSessionService(MinecraftSessionService sessionServiceIn) {
+		sessionService = sessionServiceIn;
+	}
+
+	@Override
+	public CompoundNBT write(CompoundNBT tag) {
+		super.write(tag);
+		if(!StringUtils.isNullOrEmpty(getOwner().getName()) && !(getOwner().getName().equals("owner")))
+		{
+			if(ownerProfile == null || !getOwner().getName().equals(ownerProfile.getName()))
+				setPlayerProfile(new GameProfile((UUID)null, getOwner().getName()));
```

Comment:
As partially discussed in Discord, this is the way things are handled in Vanilla's SkullTileEntity. The UUID can be null because it's initally not used and is later auto-filled in the `getGameProfileForUsername` method (line 140), which fills profile by first checking local profileCache, and if unsuccessful/uncached, by calling Mojang's servers. Passing an invalid UUID though could lead to failed authentication. We could pass in here the `getOwner().getUUID()`, but it won't provide any advantage and it could create problems if by any reason the UUID is invalid.

Let me know if you want to change this, but I'd personally leave it this way.